### PR TITLE
Add shard and cluster support

### PR DESCRIPTION
**Pull request by stephanepham-sib**

### CODE REVIEW

File: templates/macros.xml.j2 [hunk 1]
```diff
@@ -2,38 +2,9 @@
...
-        <shard>08</shard>
-        {% endif %}
-        {% if inventory_hostname in groups['clickhouse_shard09'] %}
-        <shard>09</shard>
-        {% endif %}
-        {% if inventory_hostname in groups['clickhouse_shard10'] %}
-        <shard>10</shard>
-        {% endif %}
+        <cluster>{{ cluster }}</cluster>
+        <shard>{{ shard }}</shard>
```
Comment (SananGuliyev) [inline templates/macros.xml.j2:6]: should not we have for loop here?

File: templates/instances.xml.j2 [hunk 2]
```diff
@@ -1,118 +1,24 @@
...
             <shard>
             <internal_replication>true</internal_replication>
-            {% for node_hostname in groups['clickhouse_ip-shard01'] %}
+{% for replica in replicas %}
                 <replica>
-                    <host>{{ node_hostname }}</host>
-                    <port>{{ clickhouse_tcp_port }}</port>
-                    <user>app_sib</user>
+                    <host>{{ replica['host'] }}</host>
+                    <host>{{ replica['port'] }}</host>
```
Comment (SananGuliyev) [inline templates/instances.xml.j2:14]: Is it by purpose to remove `<port>`>?

File: templates/macros.xml.j2 [hunk 3]
```diff
@@ -2,38 +2,9 @@
...
         <replica>{{ inventory_hostname }}</replica>
-        <distributed_clusters>distributed_cluster</distributed_clusters>
+        <distributed_clusters>{{ distributed_cluster }}</distributed_clusters>
```
Comment (SananGuliyev) [inline templates/macros.xml.j2:8]: was it working before and now we want to make it dynamic?

File: templates/instances.xml.j2 (hunk 2 above)
Comment (stephanepham-sib) [reply templates/instances.xml.j2]: thanks, I added again port!

File: templates/macros.xml.j2 (hunk 1 above)
Comment (stephanepham-sib) [reply templates/macros.xml.j2:6]: yes we could do something like
```
{% for k, v in macros.items() %}
    <{{ k }}>{{ v }}</{{ k }}>
{% endfor %}
```
Then in host_vars:

```
macros:
  key1: value1
  key2: value2
```
WDYT?

File: templates/macros.xml.j2 (hunk 3 above)
Comment (stephanepham-sib) [reply templates/macros.xml.j2:8]: yes correct

**Pull request merged**